genesyscloud/routing_utilization_label/resource_genesyscloud_routing_utilization_label_schema.go

```diff
@@ -0,0 +1,108 @@
+package routing_utilization_label
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mypurecloud/platform-client-sdk-go/v130/platformclientv2"
+	"strings"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	registrar "terraform-provider-genesyscloud/genesyscloud/resource_register"
+)
+
+const resourceName = "genesyscloud_routing_utilization_label"
+
+// SetRegistrar registers all the resources and exporters in the package
+func SetRegistrar(regInstance registrar.Registrar) {
+	regInstance.RegisterResource(resourceName, ResourceRoutingUtilizationLabel())
+	regInstance.RegisterExporter(resourceName, RoutingUtilizationLabelExporter())
+}
+
+func ResourceRoutingUtilizationLabel() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Routing Utilization Label. This resource is not yet widely available. Only use it if the feature is enabled.",
+
+		CreateContext: provider.CreateWithPooledClient(createRoutingUtilizationLabel),
+		ReadContext:   provider.ReadWithPooledClient(readRoutingUtilizationLabel),
+		UpdateContext: provider.UpdateWithPooledClient(updateRoutingUtilizationLabel),
+		DeleteContext: provider.DeleteWithPooledClient(deleteRoutingUtilizationLabel),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Label name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func dataSourceRoutingUtilizationLabel() *schema.Resource {
+	return &schema.Resource{
+		Description: "Data source for Genesys Cloud Routing Utilization Labels. Select a label by name.",
+		ReadContext: provider.ReadWithPooledClient(dataSourceRoutingUtilizationLabelRead),
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description:  "Label name.",
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringDoesNotContainAny("*"),
+				Required:     true,
+			},
+		},
+	}
+}
+
+func RoutingUtilizationLabelExporter() *resourceExporter.ResourceExporter {
+	return &resourceExporter.ResourceExporter{
+		GetResourcesFunc: provider.GetAllWithPooledClient(getAllRoutingUtilizationLabels),
+	}
+}
+
+func GenerateRoutingUtilizationLabelResource(resourceID string, name string, dependsOnResource string) string {
+	dependsOn := ""
+
+	if dependsOnResource != "" {
+		dependsOn = fmt.Sprintf("depends_on=[genesyscloud_routing_utilization_label.%s]", dependsOnResource)
+	}
+
+	return fmt.Sprintf(`resource "genesyscloud_routing_utilization_label" "%s" {
+		name = "%s"
+		%s
+	}
+	`, resourceID, name, dependsOn)
+}
+
+func CheckIfLabelsAreEnabled() error { // remove once the feature is globally enabled
```

Comment:
You can probably remove this comment, I assume it was from before this was refactored